### ISSUE TITLE
missing tqdm dependency

### ISSUE THREAD

**Issue by sroet** [openpathsampling/openpathsampling#915]
Running the following from the main directory on `master` fails:
```
# This will delete any existing test env, so make sure that is ok
conda create -y -n test python=3
conda activate test
conda install openpathsampling --only-deps
pip install -e .
conda install -y pytest nose
py.test
```
This fails with:
```
============================================= FAILURES ==============================================
______________________________ TestSimpleProgress.test_progress_setter ______________________________

self = <openpathsampling.tests.test_progress.TestSimpleProgress object at 0x7f909086d6d0>

    def test_progress_setter(self):
        prog = self.progress.progress
        assert prog is not None
        self.progress.progress = 'silent'
        assert self.progress.progress is silent_progress
        self.progress.progress = 'tqdm'
>       assert isinstance(self.progress.progress, TqdmPartial)
E       assert False
E        +  where False = isinstance(<function silent_progress at 0x7f90b83dc820>, TqdmPartial)
E        +    where <function silent_progress at 0x7f90b83dc820> = <openpathsampling.progress.SimpleProgress object at 0x7f909086d5e0>.progress
E        +      where <openpathsampling.progress.SimpleProgress object at 0x7f909086d5e0> = <openpathsampling.tests.test_progress.TestSimpleProgress object at 0x7f909086d6d0>.progress

openpathsampling/tests/test_progress.py:56: AssertionError
```
after running `conda install -y tqdm` it passes

(another option is to add the `pytest.importorskip(tqdm)` to that test but that seems to skip a bit too greedy)

**Comment by dwhswenson** [comment]
Actually, best is to put the lines

```python
self.progress.progress = 'tqdm'
assert isinstance(self.progress.progress, TqdmPartial)
```

(and anything else related) in an `if` block, skipping the tqdm part if tqdm isn't there. Testing for `'silent'` should happen anyway in either case. There should be some sort of `HAS_TQDM` variable defined in the test already.

There's no intention for tqdm to be anything other than optional. You get tqdm-based progress bars if tqdm is in the environment, that's all.

Tests miss this because something else in the test environment requires tqdm, so even in the minimal testing, tqdm is present.

Feel free to make a PR if you want!

**Comment by sroet** [comment]
> Feel free to make a PR if you want!

will do,

other small comment, I think #864 resolved itself (as this issue was run on python `3.8`)

**Comment by dwhswenson** [comment]
> other small comment, I think #864 resolved itself (as this issue was run on python 3.8)

Last I checked, we still can't conda install OpenMM on py38, so our testing would be limited. I think OpenMM/py38 is waiting on them getting the conda-forge release of OpenMM working (it is still in the `testing` label), at which point the omnia channel will be deprecated.

**Comment by dwhswenson** [comment]
Closed by #916.
